### PR TITLE
saineighbor.h: Updated SAI_NEIGHBOR_ENTRY_ATTR_ENCAP_INDEX and deprecated SAI_NEIGHBOR_ENTRY_ATTR_ENCAP_IMPOSE_INDEX

### PR DESCRIPTION
**Pull request by SarathBug**
Fixes https://github.com/opencomputeproject/SAI/issues/1165

Signed-off-by: SarathBug <sarath.gadagottu@broadcom.com>

### CODE REVIEW

File: inc/saineighbor.h [hunk 1]
```diff
@@ -123,13 +123,14 @@ typedef enum _sai_neighbor_entry_attr_t
...
      * @flags CREATE_AND_SET
-     * @default 0
+     * @default internal
      */
     SAI_NEIGHBOR_ENTRY_ATTR_ENCAP_INDEX,
 
     /**
      * @brief Encapsulation index is imposed
      *
+     * This attribute is deprecated
```
Comment (kcudnik) [inline inc/saineighbor.h:133]: there is a special flag that specifies that attribute is depreacated

**Review by kcudnik — CHANGES_REQUESTED** [review]

**Comment by kcudnik** [comment]
please relax saisanitycheck.c condition on 1178, add case for this scenario

File: inc/saineighbor.h (hunk 1 above)
Comment (SarathBug) [reply inc/saineighbor.h:133]: Thanks, Made that change

**Review by kcudnik — APPROVED** [review]

**Review by eswaranb — APPROVED** [review]

**Comment by SarathBug** [comment]
@rlhui 

Can this be committed if there are no other comments?

**Review by skeesara-nokia — APPROVED** [review]

**Comment by rlhui** [comment]
the PR title does not match the change - "SAI_NEIGHBOR_ENTRY_ATTR_ENCAP_INDEX " was repeated?

**Review by rlhui — CHANGES_REQUESTED** [review]
Can the PR title be corrected?

**Comment by SarathBug** [comment]
> Can the PR title be corrected?

@rlhui, Thanks for the comment. Corrected it.

**Review by rlhui — APPROVED** [review]

**Pull request merged**